spec/rubocop/cop/lint/memoized_instance_variable_name_spec.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Lint::MemoizedInstanceVariableName do
+  subject(:cop) { described_class.new }
+
+  context 'memoized variable does not match method name' do
+    it 'registers an offense' do
+      expect_offense(<<-RUBY.strip_indent)
+      def x
+        @y ||= :foo
+        ^^^^^^^^^^^ Memoized variable does not match method name.
```

Comment:
Underlining the whole first line of offense seems too much. `add_offense` takes a `location:` named parameter that accepts a range.  You may not need it though. Maybe try passing your `variable` node instead of the `first_in_method`.